actions/actions.go

```diff
@@ -203,10 +203,148 @@ func TeardownSuite(
 		log.Warn().Str("Invalid Keep Value", config.ProjectFrameworkSettings.KeepEnvironments).
 			Msg("Invalid 'keep_environments' value, see the 'framework.yaml' file")
 	}
+
+	if err := returnFunds(chainlinkNodes, nets); err != nil {
+		log.Error().Err(err).Str("Namespace", env.Namespace).
+			Msg("Error attempting to return funds from chainlink nodes to network's default wallet. " +
+				"Environment is left running so you can try manually!")
+		env.Persistent = true
+	} else {
+		log.Info().Msg("Successfully returned funds from chainlink nodes to default network wallets")
+	}
+	if nets != nil {
+		if err := nets.Teardown(); err != nil {
+			return err
+		}
+	}
 	if !env.Config.Persistent {
 		if err := env.Teardown(); err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+// Returns all the funds from the chainlink nodes to the networks default address
+func returnFunds(chainlinkNodes []client.Chainlink, networks *client.Networks) error {
+	if networks == nil {
+		log.Warn().Msg("No network connections found, unable to return funds from chainlink nodes.")
+	}
+	log.Info().Msg("Attempting to return Chainlink node funds to default network wallets")
+	for _, network := range networks.AllNetworks() {
+		if network.GetNetworkType() == client.SimulatedEthNetwork {
+			log.Info().Str("Network Name", network.GetNetworkName()).Msg("Network is a `eth_simulated` network. Skipping fund return.")
+			continue
+		}
+		addressMap, err := sendFunds(chainlinkNodes, network)
+		if err != nil {
+			return err
+		}
+
+		err = checkFunds(chainlinkNodes, addressMap, strings.ToLower(network.GetDefaultWallet().Address()))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Requests that all the chainlink nodes send their funds back to the network's default wallet
+// This is surprisingly tricky, and fairly annoying due to Go's lack of syntactic sugar and how chainlink nodes handle txs
+func sendFunds(chainlinkNodes []client.Chainlink, network client.BlockchainClient) (map[int]string, error) {
+	chainlinkTransactionAddresses := make(map[int]string)
+	sendFundsErrGroup := new(errgroup.Group)
+	for ni, n := range chainlinkNodes {
+		nodeIndex := ni // https://golang.org/doc/faq#closures_and_goroutines
+		node := n
+		// Send async request to each chainlink node to send a transaction back to the network default wallet
+		sendFundsErrGroup.Go(
+			func() error {
+				primaryEthKeyData, err := node.ReadPrimaryETHKey()
+				if err != nil {
+					return err
+				}
+				nodeBalanceString := primaryEthKeyData.Attributes.ETHBalance
+				if nodeBalanceString != "0" { // If key has a non-zero balance, attempt to transfer it back
+					gasCost, err := network.EstimateTransactionGasCost()
```

Comment:
Why do we care about gas costs here at all? Isn't standard gas estimation for EVM compatible clients is enough?